internal/pluginloader.go

```diff
@@ -398,3 +450,17 @@ func loadPluginFunction(c px.Context, n string, he hieraapi.Entry) (fn px.Functi
 	})
 	return
 }
+
+func extractOptFromContext(c px.Context, key string) string {
+	pl, ok := c.DefiningLoader().(*pluginLoader)
+	if !ok {
+		return ""
+	}
+
+	socketDir, ok := pl.he.OptionsMap()[key]
```

Comment:
Given that `extractOptFromContext` is a generic function, it's a bit misleading to name the option variable `socketDir`.

Reply:
Right, missed this one. Thanks.